### PR TITLE
Fix the Fedora dev Dockerfile.

### PR DESCRIPTION
**Pull request by dceara**
#### What this PR does and why is it needed
This PR fixes the dev Dockerfile.  It's needed in order to be able to easily test custom OVN changes.

#### Description for the changelog
Fix OVS/OVN build steps in Dockerfile.fedora.dev.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: dist/images/Dockerfile.fedora.dev [hunk 1]
```diff
@@ -49,13 +43,15 @@ RUN rm rpm/rpmbuild/RPMS/x86_64/*devel*
 RUN git log -n 1
 
 # Clone OVN Source Code.
-ARG OVN_REPO=https://github.com/ovn-org/ovn.git
-ARG OVN_BRANCH=main
+ARG OVN_REPO=https://github.com/dceara/ovn.git
```
Comment (igsilya) [inline dist/images/Dockerfile.fedora.dev:46]: Testing leftover.

Comment (tssurya) [inline dist/images/Dockerfile.fedora.dev:46]: do we want to pin this to your repo and that branch?

Comment (dceara) [reply dist/images/Dockerfile.fedora.dev:46]: :grin: oops, will fix it.

Comment (dceara) [reply dist/images/Dockerfile.fedora.dev:46]: Nope, my bad.

File: dist/images/Dockerfile.fedora.dev [hunk 2]
```diff
@@ -22,25 +22,19 @@ USER root
...
-    libpcap hostname util-linux\
-    python3-openvswitch python3-pyOpenSSL \
-    autoconf automake libtool g++ gcc fedora-packager rpmdevtools \
-    unbound unbound-devel groff python3-sphinx graphviz openssl openssl-devel \
-    libbpf-devel libxdp-devel numactl-devel \
-    checkpolicy libcap-ng-devel selinux-policy-devel" && \
-    dnf install --best --refresh -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
-    dnf clean all && rm -rf /var/cache/dnf/*
+RUN INSTALL_PKGS="git rpm-build dnf-plugins-core" && \
+    dnf install --best --refresh -y --setopt=tsflags=nodocs $INSTALL_PKGS
```
[the remark this answers is not in the document]
Comment (dceara) [reply dist/images/Dockerfile.fedora.dev:26]: Not really indeed, we're just copying the rpms from the builder in the end.  I'll remove it.

**Review by pperiyasamy — APPROVED** [review]

**Review by tssurya — CHANGES_REQUESTED** [review]
main changes lgtm
now that we are mandating docs, could you please add something to https://ovn-kubernetes.io/developer-guide/local_testing_guide/ (currently this has nothing sorry about that, but please add a subsection that says "Testing custom OVN and OVS Fixes" or something -> nothing too extravagant it would be good to document that we actually do have a dev dockerfile just for this!)

**Comment by dceara** [comment]
> main changes lgtm now that we are mandating docs, could you please add something to https://ovn-kubernetes.io/developer-guide/local_testing_guide/ (currently this has nothing sorry about that, but please add a subsection that says "Testing custom OVN and OVS Fixes" or something -> nothing too extravagant it would be good to document that we actually do have a dev dockerfile just for this and here is how you can use this guys!!!)

I can but, to be completely fair, I think the right way to do this is to open an issue (or more because as you point out there's nothing at all about testing locally) instead of adding documentation unrelated to the change (which is to fix the Dockerfile itself).

Another reason for doing that as a separate issue is that I'd have to add a subsection that talks about starting kind, e.g., `kind.sh --ovn-image <IMAGE>` but then there's nothing at all about how to use kind in general. I wouldn't be surprised if whenever the rest of the local_testing_guide is written the subsection I'd be writing now would have to change.

**Comment by tssurya** [comment]
> I could but, to be completely fair, I think the right way to do this is to open an issue (or more because as you point out there's nothing at all about testing locally) instead of adding documentation unrelated to the change (which is to fix the Dockerfile itself).

not asking for writing the entire doc, only the piece you have modified currently which is how to easily run custom ovn/ovs changes in KIND.

> 
> Another reason for doing that as a separate issue is that I'd have to add a subsection that talks about starting kind, e.g., `kind.sh --ovn-image <IMAGE>` but then there's nothing at all about how to use kind in general. I wouldn't be surprised if whenever the rest of the local_testing_guide is written the subsection I'd be writing now would have to change.

https://ovn-kubernetes.io/installation/launching-ovn-kubernetes-on-kind/ we already have it here, you can just start saying "See [here](https://ovn-kubernetes.io/installation/launching-ovn-kubernetes-on-kind/) for details on KIND installation. In order to run custom changes in OVN/OVS here are the steps:
1.
2.
3.
4.
that's all I meant like 25lines.

**Review by tssurya — APPROVED** [review]
has a +1 approver already, Dumitru doesn't have time to do docs ATM and I won't push further here. Let's just move on ~and I will write the docs for this myself~.

**Comment by dceara** [comment]
Thanks @tssurya!

**Pull request merged**